upup/pkg/fi/cloudup/openstacktasks/lb.go

```diff
@@ -93,12 +93,6 @@ func (e *LB) GetDependencies(tasks map[string]fi.Task) []fi.Task {
 		if _, ok := task.(*Subnet); ok {
 			deps = append(deps, task)
 		}
-		if _, ok := task.(*ServerGroup); ok {
-			deps = append(deps, task)
-		}
-		if _, ok := task.(*Instance); ok {
```

Comment:
ServerGroup & Instance are not used in LB task. So I do not see why these are dependencies

Reply:
I agree. This alone should break the circular dependency.